### ISSUE TITLE
Linux AppImage Error: AssertionError: undefined symbol: g_memdup2

### ISSUE THREAD

**Issue by justquick** [beeware/briefcase#812]
**Describe the bug**
Went through the tutorial with the basic hello world skeleton project. Didnt add anything extra to the toga app itself. briefcase dev works and opens a window. briefcase create && build created a linux AppImage successfully. When executing it I get the following traceback

```
$ ./linux/GUI-0.0.1-x86_64.AppImage 
** (process:47587): WARNING **: 08:52:58.884: Failed to load shared library 'libpango-1.0.so.0' referenced by the typelib: /lib/x86_64-linux-gnu/libpango-1.0.so.0: undefined symbol: g_memdup2
Traceback (most recent call last):
  File "/tmp/.mount_GUI-0.5nUnSK/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/tmp/.mount_GUI-0.5nUnSK/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app/gui/__main__.py", line 4, in <module>
    main().main_loop()
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app/gui/app.py", line 27, in main
    return GUI()
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app_packages/toga/app.py", line 278, in __init__
    self.factory = get_platform_factory(factory)
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app_packages/toga/platform.py", line 41, in get_platform_factory
    from toga_gtk import factory
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app_packages/toga_gtk/factory.py", line 1, in <module>
    from .app import App, DocumentApp, MainWindow
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app_packages/toga_gtk/app.py", line 13, in <module>
    from .keys import gtk_accel
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app_packages/toga_gtk/keys.py", line 3, in <module>
    from .libs import Gdk
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app_packages/toga_gtk/libs/__init__.py", line 1, in <module>
    from .gtk import *  # noqa: F401, F403
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app_packages/toga_gtk/libs/gtk.py", line 5, in <module>
    from gi.repository import Gtk, Gdk, GdkPixbuf, Gio, GLib, GObject  # noqa: F401, E402
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 672, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 632, in _load_backward_compatible
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app_packages/gi/importer.py", line 149, in load_module
    importlib.import_module('gi.repository.' + dep.split("-")[0])
  File "/tmp/.mount_GUI-0.5nUnSK/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 672, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 632, in _load_backward_compatible
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app_packages/gi/importer.py", line 149, in load_module
    importlib.import_module('gi.repository.' + dep.split("-")[0])
  File "/tmp/.mount_GUI-0.5nUnSK/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 672, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 632, in _load_backward_compatible
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app_packages/gi/importer.py", line 150, in load_module
    dynamic_module = load_overrides(introspection_module)
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app_packages/gi/overrides/__init__.py", line 118, in load_overrides
    override_mod = importlib.import_module(override_package_name)
  File "/tmp/.mount_GUI-0.5nUnSK/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app_packages/gi/overrides/Pango.py", line 41, in <module>
    FontDescription = override(FontDescription)
  File "/tmp/.mount_GUI-0.5nUnSK/usr/app_packages/gi/overrides/__init__.py", line 195, in override
    assert g_type != TYPE_NONE
AssertionError
```

**To Reproduce**
Just going through the tutorial: https://docs.beeware.org/en/latest/tutorial/tutorial-1.html

**Expected behavior**
The hello world window will open like it does when running 

**Environment:**
```
$ python --version
Python 3.10.4
$ briefcase --version
0.3.8
$ uname -a
Linux huawei 5.15.0-43-generic #46-Ubuntu SMP Tue Jul 12 10:30:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
$ cat /etc/issue
Ubuntu 22.04.1 LTS \n \l
```

**Comment by freakboy3742** [comment]
@justquick Thanks for the report. I'm *almost* certain this is a duplicate of one of the class of issues raised by #718, which was fixed in #756. The changes from #756 haven't made it to a formal release, but they have landed in main; could I ask if you could install the development version of Briefcase (`pip install git+https://github.com/beeware/briefcase.git`) and see if that addresses the problem?

**Comment by justquick** [comment]
Hey @freakboy3742 great talking to you again. Funny it was 10 years ago that we did the Djangocon 2012 sprints in DC together.

I had to rebuild the venv and then got the message about moving the `~/.briefcase` folder so i just redownloaded everything. It's working now thanks
